docs: secret store advantages (#279)
arcus-azure/arcus.security: docs/preview/features/secret-store/index.md

> :bulb: See [this section](#using-secret-store-within-azure-functions) if you want to use the secret store functionality whitin Azure Functions.

## Why would I use it?
Why would you use our Arcus secret store instead of just using the Azure SDK directly to access Azure Key Vault secrets?

> The Arcus secret store has some advantages over using the Azure SDK or `IConfiguration` directly:

Reviewer: Nit
```suggestion

The Arcus secret store has some advantages over using the Azure SDK or configuration directly:
```

> The Arcus secret store has some advantages over using the Azure SDK or `IConfiguration` directly:

Reviewer: I would add "design for security" so that it's clear in applications that we are working with secrets and we have to be careful, WDYT?

Context: https://github.com/dotnet/runtime/issues/36035

> The Arcus secret store has some advantages over using the Azure SDK or `IConfiguration` directly:

Reviewer: I would add "plug & play" so that you can mix and match what you want?

Author: Hmm, isn't somewhat the same as extensibility? I changed the multiple locations somewhat to include this.

> ✔ **Caching**: we provide caching so the secret providers will not be called upon every secret retrieval.

Reviewer: ```suggestion
✔ **Caching**: We provide caching so the secret providers will not be called upon every secret retrieval. This helps you avoiding hitting service limitations and we provide [asynchronous cache invalidation](https://background-jobs.arcus-azure.net/features/security/auto-invalidate-secrets).
```

> ✔ **Multiple locations**: we support multiple secret providers so with a single secret retrieval can multiple secret providers be called (also multiple Azure Key Vaults).

Reviewer: ```suggestion
✔ **Multiple locations**: We support using multiple secret providers so with a single secret retrieval can query multiple secret providers (also multiple Azure Key Vaults).
```

> ✔ **Extensibility**: the Arcus secret store is highly extensible and can be extended with you own custom secret providers, in-memory secret providers for testing...

Reviewer: ```suggestion
✔ **Extensibility**: Arcus secret store is highly extensible and can be extended with you own custom secret providers, in-memory secret providers for testing...
```

> ✔ **Security**: ex: placing secrets in the `IConfiguration` can be good for development, but it's not a safe approach. With the secret store, we provide a single place to retrieve secrets instead of the functionality is scattered across the application. 

Reviewer: What do you think of this?
```suggestion
✔ **Security**: While using configuration for storing secrets can be good for development it is not a safe approach. With the secret store, we provide a single place to retrieve secrets instead of scattering the integration across the application. 
```

> :bulb: See [this section](#using-secret-store-within-azure-functions) if you want to use the secret store functionality whitin Azure Functions.

## Why would I use it?
Why would you use our Arcus secret store instead of just using the Azure SDK directly to access Azure Key Vault secrets?

The Arcus secret store has some advantages over using the Azure SDK or configuration directly:

✔ **Caching**: We provide caching so the secret providers will not be called upon every secret retrieval. This helps you avoiding hitting service limitations and we provide [asynchronous cache invalidation](https://background-jobs.arcus-azure.net/features/security/auto-invalidate-secrets).

✔ **Plug & play**: We support using multiple and combinations of secret providers so with a single secret retrieval can query multiple secret providers (also multiple Azure Key Vaults).

✔ **Design for security**: 

> - While using configuration for storing secrets can be good for development it is not a safe approach. With the secret store, we provide a single place to retrieve secrets instead of scattering the integration across the application. 

Reviewer: Nit - Either we use bullet point for all or not, but I like this change

Author: Yes, I change this to bullet points.

Reviewer: My suggestion was more to align the rest to be consistent

Author: Ah, like that. Yeah, that would be better. Then we can still use ✔, which I kinda like in this context 😄 .
